src/vs/workbench/contrib/interactiveEditor/browser/interactiveEditorWidget.ts

```diff
@@ -461,8 +463,28 @@ export class InteractiveEditorWidget {
 		this._previewDiffEditor.getOriginalEditor().setHiddenAreas(hiddenOriginal.map(lineRangeAsRange), 'diff-hidden');
 		this._previewDiffEditor.getModifiedEditor().setHiddenAreas(hiddenModified.map(lineRangeAsRange), 'diff-hidden');
 		this._previewDiffEditor.revealLine(modifiedLineRange.startLineNumber, ScrollType.Immediate);
-
 		this._onDidChangeHeight.fire();
+		if (this._accessibilityService.isScreenReaderOptimized()) {
+			this._previewDiffEditor.onDidUpdateDiff(() => {
```

Comment:
This listener leaks. Also I quite don't understand why this need to be done on event not when setting in diff input.

Reply:
I couldn't find the right place for this, but am aware it's wrong/ a hack. Will investigate

Reply:
this is needed so the screen reader user doesn't have either tediously navigate the original vs modified editors or invoked f7 every time. we are doing that for them